middleware.go

```diff
@@ -200,6 +201,32 @@ func Session(cfg Options) func(next http.Handler) http.Handler {
 				if projectClaim > 0 {
 					ctx = WithProjectID(ctx, uint64(projectClaim))
 					sessionType = max(sessionType, proto.SessionType_Project)
+				} else if projectIDClaim > 0 {
+					ctx = WithProjectID(ctx, uint64(projectIDClaim))
+					sessionType = max(sessionType, proto.SessionType_Project)
+				}
+
+				// Restrict CORS for Builder Admin API Secret Keys.
+				// These keys are designed for backend service use by third-party customers, not for web apps.
+				if accountClaim != "" && projectClaim > 0 {
+					// Secret Keys are distinguished from Wallet JWTs or Builder session JWTs
+					// by the presence of both `project` and `account` claims. (As of Dec '24)
+					// Related discussion: https://github.com/0xsequence/issue-tracker/issues/3802.
+
+					origin := r.Header.Get("Origin")
+					if origin != "" {
+						err := proto.ErrSecretKeyCorsDisallowed.WithCausef("project_id: %v", projectClaim)
+
+						slog.ErrorContext(ctx, "CORS disallowed for Secret Key",
+							slog.Any("error", err),
+							slog.String("origin", origin),
+							slog.Uint64("project_id", uint64(projectClaim)),
+						)
+
+						// TODO: Uncomment once we're confident it won't disrupt major customers.
+						// cfg.ErrHandler(r, w, err)
+						// return
+					}
```

Comment:
For now, this will only log error with `origin` and `project_id` fields, if we encounter a Secret Key misused from a web app.

Eventually, we will error out instead, once we're confident it won't disrupt major customers.